### PR TITLE
added indexname to the toggle request interface

### PR DESCRIPTION
**Pull request by sam0044**
Currently, there is a bug in the app where if you try to toggle off/on a specialization, the fields Document Count, Strictness, Past messages included and MaxTokenResponse revert back to default. The reason this is happening is because when we toggle the specialization, qSpecializationmutate only has the isActive field set and everything else is null. Because of that when we create the response, since the index name is null, the aforementioned fields are set to null because these fields should not be set without an index. This behavior is somewhat expected because we have a 'None' field in index name because of general being a specialization. So I am passing the index name in the Toggle request interface to distinguish between index being updated and just a specialization toggle.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by wdavidcamden** [comment]
Can we update this to use a patch methodology? if only isActive is passed then that is the only thing that should be changed. Not sure why we need to pass along the index name...

**Comment by sam0044** [comment]
@wdavidcamden We are sending a patch request. There is conditional logic for the request we are sending that checks whether in the request if the index is null and it is in the case of toggle, so it sets the fields to null based on that. The logic is there because since we made general a specialization, index can be removed for a specialization so the logic can't tell whether index is null because we are updating it to be or if its because of the toggle. So when we get the response back, all the specialization fields are set besides the ones that were set to null because of the conditional logic and then in the frontend, we have values that those fields revert to if they are null. Thats why I am passing the index name along. I attached the link for the code for the conditional logic below. If you have suggestions for something better, I can do that.

https://github.com/Quartech/chat-copilot/blob/7c25c1c78aa4a6a41a8e67745413156c682f890d/webapi/Services/QSpecializationService.cs#L125

**Comment by wdavidcamden** [comment]
Can you update the backend logic so there is a difference between explicitly setting the field to Null and not having the field in the patch request? so if the field is not passed in, its ignored. if its passed in but set to null then it processes the request and updates the data.

**Comment by sam0044** [comment]
@wdavidcamden I am not sure that can be done with the way how everything is working. When we send a patch request a QspecializationMutate form is made and all the fields are set before we call the UpdateSpecialization method with the form. I don't think there is a way for me to check in between. Thats why I passed the index name because thats one way I thought we could let it know that we didn't remove it explicitly. @GrahamS-Quartech Can you confirm this?

**Comment by wdavidcamden** [comment]
I was thinking something along this: https://stackoverflow.com/questions/76221332/differentiate-between-skipped-value-and-null-value-in-json

**Comment by sam0044** [comment]
@wdavidcamden I tried it. I am having issues with getting the value for the index when it is actually changed or updated. I can ask someone on the team to see if I am missing something.

**Comment by GrahamS-Quartech** [comment]
Alternatively, you could just modify the logic so that these document fields being set does not depend on IndexName being set. You'd have to confirm that those fields being present doesn't mess with a general specialization, but I don't think it would.

**Comment by sam0044** [comment]
@wdavidcamden I implemented what you wanted based on the stackoverflow post and removed indexname from the patch request

File: webapi/Controllers/SpecializationIndexController.cs
Comment (wdavidcamden) [inline webapi/Controllers/SpecializationIndexController.cs:1]: If we converted QSpecializationIndexMutate  to JsonPatchDocument<QSpecializationIndexMutate> could we just use JsonPatchDocument.ApplyTo in the service for applying the changes?

Comment (sam0044) [reply webapi/Controllers/SpecializationIndexController.cs:1]: @wdavidcamden Do you mean QSpecializationBase file? But if make changes to that, would we wanna make changes to all the mutate files and use document patch instead? We might have to change a few other things for that as well. Would it make sense to do that as part of another ticket?

**Review by wdavidcamden — APPROVED** [review]

**Pull request merged**